### PR TITLE
Easystroke

### PR DESCRIPTION
**Pull request by gtryujb**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl
  - [x] x86_64
  - [x] i686

### CODE REVIEW

File: srcpkgs/easystroke/template [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Template file for 'easystroke'
+pkgname=easystroke
+version=0.6.0
+revision=1
+archs="i686 x86_64"
```
Comment (ericonr) [inline srcpkgs/easystroke/template:5]: No reason to be restricted.

File: srcpkgs/easystroke/template [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+wrksrc=patched-easystroke-master
+build_wrksrc=easystroke
+build_style=gnu-makefile
+hostmakedepends="pkg-config"
+makedepends="boost-devel dbus-glib-devel gtkmm-devel xorg-server-devel"
+short_desc="Gesture recognition application for X11"
+maintainer="lopid <github23429@rainslide.net>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/thjaeger/easystroke/wiki/"
+distfiles="http://openartisthq.org/easystroke/patched-easystroke-master.tar.bz2"
```
Comment (ericonr) [inline srcpkgs/easystroke/template:15]: We don't want packages using unversioned tarballs. This means any update from upstream breaks our build. Where did you get the version number from?

File: srcpkgs/easystroke/template [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+checksum=18bf167eaf7c41d91e6c17b1f8436eeb0988d34d47f523ef03789caab81e902a
+tags="mouse gestures"
+
+do_build() {
+	make PREFIX=/usr
```
Comment (ericonr) [inline srcpkgs/easystroke/template:20]: Add `make_build_args="PREFIX=/usr"` to template instead.

File: srcpkgs/easystroke/patches/git.patch [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ GIT      = $(wildcard .git/index version)
+ DIST     = easystroke-$(VERSION)
+ 
+@@ -141,7 +141,6 @@
+ tmp/$(DIST): $(GIT)
+ 	$(RM) -r tmp
+ 	mkdir tmp
+-	git archive --format=tar --prefix=$(DIST)/ HEAD | (cd tmp && tar x)
+ 	echo $(VERSION) > $@/version
+ 	$(RM) $@/.gitignore $@/release
```
Comment (ericonr) [inline srcpkgs/easystroke/patches/git.patch:19]: This target shouldn't run...

File: srcpkgs/easystroke/patches/git.patch [hunk 5]
```diff
@@ -0,0 +1,20 @@
+diff -urb patched-easystroke-master.orig/easystroke/Makefile patched-easystroke-master/easystroke/Makefile
+--- patched-easystroke-master.orig/easystroke/Makefile	2021-07-19 17:16:39.949855973 +0100
++++ patched-easystroke-master/easystroke/Makefile	2021-07-19 17:17:43.041490580 +0100
+@@ -45,7 +45,7 @@
+ GENFILES = gui.c desktop.c po/POTFILES.in easystroke.desktop
+ GZFILES  = $(wildcard *.gz)
+ 
+-VERSION  = $(shell test -e debian/changelog && grep '(.*)' debian/changelog | sed 's/.*(//' | sed 's/).*//' | head -n1 || (test -e version && cat version || git describe))
++VERSION  =
```
Comment (ericonr) [inline srcpkgs/easystroke/patches/git.patch:9]: Just add `VERSION=${version}` to `make_build_args`; this allows us to completely avoid the patch.

**Comment by Anachron** [comment]
- Remove the merge commit
- Replace distfiles with `https://github.com/thjaeger/easystroke/archive/refs/tags/${version}.tar.gz`
- Fix the suggestions from @ericonr

File: srcpkgs/easystroke/template (hunk 3 above)
Comment (gtryujb) [reply srcpkgs/easystroke/template:20]: It won't build with that. Here's the difference, with your suggestion on the left: 
Screenshot_2021-07-25_15-00-16

**Comment by paper42** [comment]
@gtryujb `git pull --rebase --autostash upstream master` is the correct command to remove the merge branch commit

**Pull request closed without merging**